chore(docs): fix typos and enhance doc for better readability

rafaelsmf opened the pull request:
<!--
*Thank you very much for contributing to Fluss - we are happy that you want to help us improve Fluss. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GitHub issue](https://github.com/alibaba/fluss/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no issue.

  - Name the pull request in the format "[component] Title of the pull request", where *[component]* should be replaced by the name of the component being changed. Typically, this corresponds to the component label assigned to the issue (e.g., [kv], [log], [client], [flink]). Skip *[component]* if you are unsure about which is the best component.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.

  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes.

  - Each pull request should address only one issue, not mix up code from multiple issues.

wuchong reviewed (commented):
Hi @rafaelsmf, thanks for your contribution!

I noticed the main change in this PR is renaming `Engine Flink` to `Flink Engine`. While we appreciate the suggestion, the current naming (`Engine Flink`) is intentional.  

As we're expanding support to other engines like Spark and Trino, using a consistent prefix — such as `Engine Flink`, `Engine Spark`, `Engine Trino` — helps keep the naming structure uniform and more organized across the project.

Let’s keep it as `Engine Flink` for now to maintain consistency. Thanks again for your input!

rafaelsmf commented:
> Hi @rafaelsmf, thanks for your contribution!
> 
> I noticed the main change in this PR is renaming `Engine Flink` to `Flink Engine`. While we appreciate the suggestion, the current naming (`Engine Flink`) is intentional.
> 
> As we're expanding support to other engines like Spark and Trino, using a consistent prefix — such as `Engine Flink`, `Engine Spark`, `Engine Trino` — helps keep the naming structure uniform and more organized across the project.
> 
> Let’s keep it as `Engine Flink` for now to maintain consistency. Thanks again for your input!

Hi @wuchong ,

Thanks for the review. I now understand the structure of the Flink engine section and have reverted the proposed rename. I also fixed several typos and improved parts of the doc for readability.

Additionally, I updated the URL from Alibaba to Apache as suggested.
